### ISSUE TITLE
Long Audio - Python Sample to C Sharp Sample?

### ISSUE THREAD

**Issue by Jay-study-nildana** [Azure-Samples/cognitive-services-speech-sdk#1387]
I am trying to use the Long Audio API. For some reason, the sample in the documentation is for Python instead of C Sharp. There is a C Sharp sample linked, and I am unable to understand it. 

Here is the Python code. 

```

def submit_synthesis():
    region = '<region>'
    key = '<your_key>'
    input_file_path = '<input_file_path>'
    locale = '<locale>'
    url = 'https://{}.customvoice.api.speech.microsoft.com/api/texttospeech/v3.0/longaudiosynthesis'.format(region)
    header = {
        'Ocp-Apim-Subscription-Key': key
    }

    voice_identities = [
        {
            'voicename': '<voice_name>'
        }
    ]

    payload = {
        'displayname': 'long audio synthesis sample',
        'description': 'sample description',
        'locale': locale,
        'voices': json.dumps(voice_identities),
        'outputformat': 'riff-16khz-16bit-mono-pcm',
        'concatenateresult': True,
    }

    filename = ntpath.basename(input_file_path)
    files = {
        'script': (filename, open(input_file_path, 'rb'), 'text/plain')
    }

    response = requests.post(url, payload, headers=header, files=files)
    print('response.status_code: %d' % response.status_code)
    print(response.headers['Location'])

submit_synthesis()

```

I dont quiet understand python, but, I recognize the code to some extent. I was able to use this, and refer the linked sample, and came up with the following.

C Sharp Samples : https://github.com/Azure-Samples/Cognitive-Speech-TTS/tree/master/CustomVoice-API-Samples/CSharp

```

public async Task LongAudioTestOne()
{
    var tempStuff = new ExtraStuff();
    //path for the input text file
    var workingDirectory = Directory.GetCurrentDirectory();
    var inputFileName = "samplefile.txt";
    var tempFileName = Path.Combine(workingDirectory, inputFileName);

    string subscriptionKey = tempStuff.subscriptionKey;

    using (FileStream fstream = new FileStream(tempFileName, FileMode.Open))
    using (var client = new HttpClient())
    using (var content = new MultipartFormDataContent())
    {
        client.DefaultRequestHeaders.Add("Ocp-Apim-Subscription-Key", subscriptionKey);

        content.Add(new StringContent("sample display name"), "displayname");
        content.Add(new StringContent("sample description"), "description");
        content.Add(new StringContent("en-GB"), "locale");
        content.Add(new StringContent("[{\"voiceName\": \"en - GB - LibbyNeural\"}]"), "voices");
        content.Add(new StringContent("riff-16khz-16bit-mono-pcm"), "outputformat");

        //add file
        content.Add(new StreamContent(fstream), inputFileName, inputFileName);

        string url = @"https://centralindia.customvoice.api.speech.microsoft.com/api/texttospeech/v3.0/longaudiosynthesis";
        var response = client.PostAsync(url, content).Result;

        if (response.StatusCode != HttpStatusCode.Accepted)
        {
            //APIHelper.PrintErrorMessage(response);
            //return false;
            var something = response.StatusCode;
        }

        var something2 = response.StatusCode;
    }
}

```

I keep getting BAD REQUEST 400. I wish to know what is the mistake I am making here. 

I feel like, this specific section,

```

    files = {
        'script': (filename, open(input_file_path, 'rb'), 'text/plain')
    }

```

is where I am losing out. 

------------

On a related note, I am curious, why is there isn't a C # sample code for the same?

The linked sample project is not that easy to use, and it merges two scenarios - training and long audio, into a single function, which is hard to follow. 

Or, is this one of those things where, this simply works better on Python?

**Comment by Jay-study-nildana** [comment]
Update: 

Decided to give the Python code a try. I was able to use the Python code within seconds and get the work done. 

But, I really want to use this in C #. I hope someone can help.

**Comment by Ling-Cao** [comment]
Acknowledged and investigating.
Thanks,
Ling

**Comment by Jay-study-nildana** [comment]
@Ling-Cao 

Thank you so much. 

As of now, I am able to get my work done with the Python code, but, as our software is C Sharp, .NET that would be most useful.

**Comment by Ling-Cao** [comment]
Hello @Jay-study-nildana, I tried it and found the script file prameter format is not expected, please try the C# code with making 2 changes as below, which works well for me:
1. Change voice name from "en - GB - LibbyNeural" to "en-GB-LibbyNeural"
2. Change the add file part from:
```
//add file
content.Add(new StreamContent(fstream), inputFileName, inputFileName);
```
to (refer to [the sample code](https://github.com/Azure-Samples/Cognitive-Speech-TTS/blob/master/CustomVoice-API-Samples/CSharp/CustomVoice-API/API/BatchSynthesis.cs#:~:text=var%20scriptContent%20%3D%20new,scriptContent%2C%20%22script%22%2C%20scriptName)):
```
var scriptContent = new StreamContent(fstream);
scriptContent.Headers.Add("Content-Disposition", $@"form-data; name=""script""; filename=""{inputFileName}""");
scriptContent.Headers.Add("Content-Type", "text/plain");
scriptContent.Headers.Add("Content-Length", $"{fstream.Length}");
content.Add(scriptContent, "script", inputFileName);
```

Please let me know if the error persists.

Thanks,
Ling

**Comment by Jay-study-nildana** [comment]
@Ling-Cao 

I will check this today and update with my results. 

Once again, thank you so much. Our team were getting ready to move to Django/Python for this, but, now, hopefully, we don't have to.

**Comment by Ling-Cao** [comment]
@Jay-study-nildana 

No problem, hope it works for you, please share me if you got the result. :)

Thanks,
Ling

**Comment by Jay-study-nildana** [comment]
@Ling-Cao 

It worked like magic! I would not have been successfull without you pointing me in the right direction. 

I have put the full code here if someone else comes here looking for this. 

Note : This is NOT the most elegant piece of code, but putting here anyway :) 

#### Placing A Request for Long Audio Service

```

public async Task LongAudioTestOne()
{
    var tempStuff = new ExtraStuff();
    //path for the input text file
    var workingDirectory = Directory.GetCurrentDirectory();
    var inputFileName = "samplefile.txt";
    var tempFileName = Path.Combine(workingDirectory, inputFileName);

    string subscriptionKey = tempStuff.subscriptionKey;

    using (FileStream fstream = new FileStream(tempFileName, FileMode.Open))
    using (var client = new HttpClient())
    using (var content = new MultipartFormDataContent())
    {
        client.DefaultRequestHeaders.Add("Ocp-Apim-Subscription-Key", subscriptionKey);

        content.Add(new StringContent("sample display name"), "displayname");
        content.Add(new StringContent("sample description"), "description");
        content.Add(new StringContent("en-GB"), "locale");
        content.Add(new StringContent("[{\"voiceName\": \"en-GB-LibbyNeural\"}]"), "voices");
        content.Add(new StringContent("riff-16khz-16bit-mono-pcm"), "outputformat");

        //add file
        //content.Add(new StreamContent(fstream), inputFileName, inputFileName);

        var scriptContent = new StreamContent(fstream);
        scriptContent.Headers.Add("Content-Disposition", $@"form-data; name=""script""; filename=""{inputFileName}""");
        scriptContent.Headers.Add("Content-Type", "text/plain");
        scriptContent.Headers.Add("Content-Length", $"{fstream.Length}");
        content.Add(scriptContent, "script", inputFileName);

        string url = @"https://centralindia.customvoice.api.speech.microsoft.com/api/texttospeech/v3.0/longaudiosynthesis";
        var response = client.PostAsync(url, content).Result;

        if (response.StatusCode != HttpStatusCode.Accepted)
        {
            //APIHelper.PrintErrorMessage(response);
            //return false;
            var something = response.StatusCode;
        }

        var something2 = response.StatusCode;
    }
}

```
#### Checking for status
```
public async Task CheckStatus(string tempURLOfStatus)
{
    var tempStuff = new ExtraStuff();
    string subscriptionKey = tempStuff.subscriptionKey;
    using (var client = new HttpClient())
    {
        client.DefaultRequestHeaders.Add("Ocp-Apim-Subscription-Key", subscriptionKey);

        string url = tempURLOfStatus;
        //var response = client.PostAsync(url, content).Result;
        var response = await client.GetAsync(url);
        var responseStringJSON = await client.GetStringAsync(url);

        if (response.StatusCode != HttpStatusCode.Accepted)
        {
            //APIHelper.PrintErrorMessage(response);
            //return false;
            var something = response.StatusCode;
            var tempJSON = response.ToString();
            string responseBody = await response.Content.ReadAsStringAsync();

            string jsonString = responseBody;
            var longAudioStatus = LongAudioStatus.FromJson(jsonString);
        }

        var something2 = response.StatusCode;
    }
}
```
#### Getting the Files
```
public async Task GetAudioResult(string tempRequestID)
{
    var tempStuff = new ExtraStuff();
    string subscriptionKey = tempStuff.subscriptionKey;
    using (var client = new HttpClient())
    {
        client.DefaultRequestHeaders.Add("Ocp-Apim-Subscription-Key", subscriptionKey);

        string url = tempRequestID;
        //var response = client.PostAsync(url, content).Result;
        var response = await client.GetAsync(url);
        var responseStringJSON = await client.GetStringAsync(url);

        if (response.StatusCode != HttpStatusCode.Accepted)
        {
            //APIHelper.PrintErrorMessage(response);
            //return false;
            var something = response.StatusCode;
            var tempJSON = response.ToString();
            string responseBody = await response.Content.ReadAsStringAsync();

            string jsonString = responseBody;
            var longAudioStatusFile = LongAudioStatusFile.FromJson(jsonString);
        }

        var something2 = response.StatusCode;
    }
}
```

Again, not the most elegant code.

**Comment by Ling-Cao** [comment]
@Jay-study-nildana, Thank you for sharing, I believe it would be helpful too, BTW, maybe the add file part can be removed~
```
//add file
//content.Add(new StreamContent(fstream), inputFileName, inputFileName);
```
Thanks,
Ling
